docs/guides/redis/clustering/redis-cluster.md

```diff
@@ -0,0 +1,415 @@
+-> Redirected to slot [866] located at 172.17.0.4:6379
+"world"
+172.17.0.4:6379> exit
+
+# switch the connection to any other node
+# get the data
+/data # redis-cli -c -h 172.17.0.11
+172.17.0.11:6379> get hello
+-> Redirected to slot [866] located at 172.17.0.4:6379
+"world"
+172.17.0.4:6379> exit
+```
+
+## Automatic Failover
+
+To test automatic failover, we will force a master node to restart. Since the master node (`pod`) becomes unavailable, its slave (one of its slave in case of more than one slave under this master) the rest of the members will elect a slave of this master node as the new master. When the old master comes back, it will join the cluster as the new slave of the new master.
+
+> Read the comment written for the following commands. They contain the instructions and explanations of the commands.
+
+```console
+# connect to any node and get the master nodes info
+$ kubectl exec -it redis-cluster-shard0-0 -n demo -c redis -- sh
+/data # redis-cli -c cluster nodes | grep master
+e49d748b815db355f29e670ba62b669627384273 172.17.0.10:6379@16379 master - 0 1550589457000 2 connected 5461-10922
+0cfa026b921bd07c36a95734edf5ccd73cda5d50 172.17.0.12:6379@16379 master - 0 1550589457739 3 connected 10923-16383
+37ae24f6f2442cefc34fc5d3c678b2aff8f13d26 172.17.0.4:6379@16379 myself,master - 0 1550589456000 1 connected 0-5460
+
+# let's crash node 172.17.0.4 with the `DEBUG SEGFAULT` command
+/data # redis-cli -h 172.17.0.4 debug segfault
+Error: Server closed the connection
+
+# now again connect to a node and get the master nodes info
+$ kubectl exec -it redis-cluster-shard0-0 -n demo -c redis -- sh
+/data # redis-cli -c cluster nodes | grep master
+e49d748b815db355f29e670ba62b669627384273 172.17.0.10:6379@16379 master - 0 1550589881100 2 connected 5461-10922
+4136ea1a767fd26d76ad7f8066d7eab994850048 172.17.0.8:6379@16379 master - 0 1550589880000 4 connected 0-5460
+0cfa026b921bd07c36a95734edf5ccd73cda5d50 172.17.0.12:6379@16379 master - 0 1550589881000 3 connected 10923-16383
+
+/data # redis-cli -c cluster nodes
+e425958b07698cc69e62e0f2c94f5951155cbe71 172.17.0.11:6379@16379 slave e49d748b815db355f29e670ba62b669627384273 0 1550590186590 2 connected
+e49d748b815db355f29e670ba62b669627384273 172.17.0.10:6379@16379 master - 0 1550590186990 2 connected 5461-10922
+981f26c1e2d16f56109ca74ee79aaa5cd5e62a79 172.17.0.13:6379@16379 slave 0cfa026b921bd07c36a95734edf5ccd73cda5d50 0 1550590186000 3 connected
+37ae24f6f2442cefc34fc5d3c678b2aff8f13d26 172.17.0.4:6379@16379 myself,slave 4136ea1a767fd26d76ad7f8066d7eab994850048 0 1550590185000 1 connected
+4136ea1a767fd26d76ad7f8066d7eab994850048 172.17.0.8:6379@16379 master - 0 1550590184585 4 connected 0-5460
+0cfa026b921bd07c36a95734edf5ccd73cda5d50 172.17.0.12:6379@16379 master - 0 1550590186000 3 connected 10923-16383
+
+/data # exit
+```
+
+Notice that 172.17.0.8 is the new master and  172.17.0.4 is the slave of  172.17.0.8.
+
+## Delete Redis and WipeOut DormantDatabase
```

Comment:
Remove this part. Already `Quickstart` doc have this information.

Add #cleanup